lib/file-source-common/src/fingerprinter.rs

```diff
@@ -201,19 +201,19 @@ impl Fingerprinter {
         path: &Path,
         known_small_files: &mut HashMap<PathBuf, time::Instant>,
         emitter: &impl FileSourceInternalEvents,
-    ) -> Option<FileFingerprint> {
-        let metadata = match fs::metadata(path).await {
```

Comment:
This being named metadata but actually being the fingerprint was definitely off. Changes in this file are very nice, improve readability a lot and should also improve performance. Great job 👍